test/com/google/javascript/jscomp/PeepholeRemoveDeadCodeTest.java

```diff
@@ -478,6 +481,226 @@ public void testOptimizeSwitch2() {
         "outer: {f(); break outer;}");
   }
 
+  public void testOptimizeSwitchWithLabellessBreak() {
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  switch('x') {",
+        "    case 'x': var x = 1; break;",
+        "    case 'y': break;",
+        "  }",
+        "}"),
+        "function f() { var x = 1; }");
+
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  switch('x') {",
+        "    case 'x': var x = 1; break; var y = 2;",
+        "    case 'y': break;",
+        "  }",
+        "}"),
+        "function f() { var x = 1; }");
+
+    testSame(LINE_JOINER.join(
+        "function f() {",
+        "  switch(x) {",
```

Comment:
For one, it can be converted to a hook `x == 'y' ? /* case block */ : /* else block */`. It can also use `&&` and `||` shorthands.